cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -1738,7 +1738,16 @@ angular.module(PKG.name + '.commons')
       vm.pluginsMap = AvailablePluginsStore.getState().plugins.pluginsMap;
       $scope.nodes.forEach(node => {
         let key = generatePluginMapKey(node);
-        node.isPluginAvailable = Boolean(myHelpers.objectQuery(vm.pluginsMap, key, 'pluginInfo')) ;
+        // This is to check if the plugin version is a range. If so, mark the plugin
+        // as available and UI will decide on the specific version while opening the plugin.
+        if (
+            myHelpers.objectQuery(node, 'plugin', 'artifact', 'version') &&
+            node.plugin.artifact.version.indexOf('[') === 0
+          ) {
+          node.isPluginAvailable = true;
```

Comment:
might need to be addressed later, but what would happen if there is no plugin available within the range?

Reply:
This should already be handled. When we open the plugin properties we fetch the properties from the backend. If that returns 404 (which it should given artifact name, version and scope together forms the unique id) we fallback to the same message.